en/lessons/libraries/distillery.md

```diff
@@ -0,0 +1,245 @@
+mix release.init
+```
+
+This command generates a rel directory with some configuration files in it.
+
+To generate a release in your terminal run `mix release`
+
+Once the release is built you should see some instructions in your terminal
+
+```
+==> Assembling release..
+==> Building release book_app:0.1.0 using environment dev
+==> You have set dev_mode to true, skipping archival phase
+Release successfully built!
+To start the release you have built, you can use one of the following tasks:
+
+    # start a shell, like 'iex -S mix'
+    > _build/dev/rel/book_app/bin/book_app console
+
+    # start in the foreground, like 'mix run --no-halt'
+    > _build/dev/rel/book_app/bin/book_app foreground
+
+    # start in the background, must be stopped with the 'stop' command
+    > _build/dev/rel/book_app/bin/book_app start
+
+If you started a release elsewhere, and wish to connect to it:
+
+    # connects a local shell to the running node
+    > _build/dev/rel/book_app/bin/book_app remote_console
+
+    # connects directly to the running node's console
+    > _build/dev/rel/book_app/bin/book_app attach
+
+For a complete listing of commands and their use:
+
+    > _build/dev/rel/book_app/bin/book_app help
+```
+
+
+
+
+To run your application type the following in your terminal ` _build/dev/rel/MYAPP/bin/MYAPP foreground`
+In your case replace MYAPP with your project name.
+
+
+## Using distillery with Phoenix
+
+If you are using distillery with Phoenix there are a few extra steps you need to follow before this will work.
+
+First, we need to edit our `config/prod.exs` file.
+
+Change the following line form this:
+
+```
+config :book_app, BookAppWeb.Endpoint,
+  load_from_system_env: true,
+  url: [host: "example.com", port: 80],
+  cache_static_manifest: "priv/static/cache_manifest.json"
+```
+to this:
+
+```
+config :book_app, BookApp.Endpoint,
+  http: [port: {:system, "PORT"}],
+  url: [host: "localhost", port: {:system, "PORT"}],
+  cache_static_manifest: "priv/static/cache_manifest.json",
+  server: true,
+  root: ".",
+  version: Application.spec(:book_app, :vsn)
+```
+
+Note
+
+We specified that the port will be passed in via an ENV variable. When we run our our application as we did above we will need to pass in the PORT
+
+`PORT=4001 _build/prod/rel/book_app/bin/book_app foreground`
+
+If you executed the following command you might have noticed that your application crashed because it is unable to connect to the databse. This is pretty straightforward to do with postgres. In your terminal, type the following:
+
+```
+sudo psql
+postgres=# CREATE DATABASE book_app_production;
+
+```
+
+Try re-running the application and it should start up successfully. However, you will notice that your migrations to your database have not run. Usually in development we run those migrations manually by calling `mix.ecto migrate`. For the release, we will have to configure it so that it can run the migrations on its own.
+
+
+## Configuring your release
```

Comment:
This title doesn't really describe what this section is about. Maybe something like: "Running Migrations in Production"